### PR TITLE
adds async function for list in Storage Plugin

### PR DESCRIPTION
**Pull request by brennanMKE**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift [hunk 1]
```diff
@@ -95,4 +95,9 @@ extension StorageCategory: StorageCategoryBehavior {
                        options: StorageRemoveRequest.Options? = nil) async throws -> String {
         try await plugin.remove(key: key, options: options)
     }
+
+    @discardableResult
+    public func list(options: StorageListOperation.Request.Options?) async throws -> StorageListResult {
```
Comment (lawmicha) [inline Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift:100]: a reminder for default parameters for `list` and `getURL`

**Review by lawmicha — APPROVED** [review]
a reminder for default parameters for `list` and `getURL`, otherwise LGTM

File: Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift (hunk 1 above)
Comment (lawmicha) [reply Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift:100]: https://github.com/aws-amplify/amplify-ios/issues/2223 for reminder

Comment (brennanMKE) [reply Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift:100]: This is the same as the functions which are being copied. I did not change it.

Comment (brennanMKE) [reply Amplify/Categories/Storage/StorageCategory+ClientBehavior.swift:100]: Here is the existing API. I just dropped the listener parameter and changes this return type.

```swift
@discardableResult
func list(options: StorageListOperation.Request.Options?,
          resultListener: StorageListOperation.ResultListener?) -> StorageListOperation
```

**Pull request merged**